pkg/orchestration/wiring/entangler.go

```diff
@@ -232,6 +234,12 @@ func (w *worker) entangle(resource *orch_v1.StateResource, stateMeta *meta_v1.Ob
 	if w.allWiredResources[resource.Name] != nil {
 		return false, errors.New("internal error in wiring plugin - duplicate resource name received from plugin")
 	}
+
+	if shapes := findDuplicateShapes(result.Contract.Shapes); shapes != nil {
```

Comment:
The one thing I don't like about that is when you the code doing the check is type-opaque (doesn't know whether the thing it's looking at is a slice or an array or a pointer or whatever). So defensively you kinda have to write thing != nil in some situations.

Reply:
> doesn't know whether the thing it's looking at

Go is a statically typed language - you always know the type. It may be an interface but comparing a variable of an interface type to `nil` might not be doing what you think it is doing judging by what you are saying =) Or maybe I'm not getting what you are saying, sorry.